pkg/eigenState/stakerShares/stakerShares.go

```diff
@@ -410,102 +449,120 @@ func parseLogOutputForSlashingWithdrawalQueuedEvent(outputDataStr string) (*slas
 	outputData, err := parseLogOutputForSlashingWithdrawalQueuedEvent(log.OutputData)
 	if err != nil {
 		return nil, err
 	}
 
-	records := make([]*StakerShareDeltas, 0)
+	records := make([]*StateDiff, 0)
 
 	for i, strategy := range outputData.Withdrawal.Strategies {
 		shares, success := numbers.NewBig257().SetString(outputData.SharesToWithdraw[i].String(), 10)
 		if !success {
 			return nil, xerrors.Errorf("Failed to convert shares to big.Int: %s", outputData.SharesToWithdraw[i])
 		}
-		r := &StakerShareDeltas{
-			Staker:          outputData.Withdrawal.Staker,
-			Strategy:        strategy,
-			Shares:          shares.Mul(shares, big.NewInt(-1)).String(),
-			StrategyIndex:   uint64(i),
-			LogIndex:        log.LogIndex,
-			TransactionHash: log.TransactionHash,
-			BlockNumber:     log.BlockNumber,
+		r := &StateDiff{
+			Event: &StakerShareDeltas{
+				Staker:          outputData.Withdrawal.Staker,
+				Strategy:        strategy,
+				Shares:          shares.Mul(shares, big.NewInt(-1)).String(),
+				StrategyIndex:   uint64(i),
+				LogIndex:        log.LogIndex,
+				TransactionHash: log.TransactionHash,
+				BlockNumber:     log.BlockNumber,
+			},
 		}
 		records = append(records, r)
 	}
 	return records, nil
 }
 
-type AccumulatedStateChanges struct {
-	Changes []*AccumulatedStateChange
+type operatorSlashedOutputData struct {
+	Operator string `json:"operator"`
+	// OperatorSet struct {
+	// 	Avs           string `json:"avs"`
+	// 	OperatorSetId uint32 `json:"operatorSetId"`
+	// }
+	Strategies  []string      `json:"strategies"`
+	WadsSlashed []json.Number `json:"wadsSlashed"`
+	// Description string        `json:"description"`
 }
 
-func shareDeltaToAccumulatedStateChange(deltaRecord *StakerShareDeltas) *AccumulatedStateChange {
-	shares, _ := numbers.NewBig257().SetString(deltaRecord.Shares, 10)
-	return &AccumulatedStateChange{
-		Staker:      deltaRecord.Staker,
-		Strategy:    deltaRecord.Strategy,
-		Shares:      shares,
-		BlockNumber: deltaRecord.BlockNumber,
+func parseLogOutputForOperatorSlashedEvent(outputDataStr string) (*operatorSlashedOutputData, error) {
+	outputData := &operatorSlashedOutputData{}
+	decoder := json.NewDecoder(strings.NewReader(outputDataStr))
+	decoder.UseNumber()
+
+	err := decoder.Decode(&outputData)
+	if err != nil {
+		return nil, err
+	}
+
+	return outputData, err
+}
+
+func (ss *StakerSharesModel) handleOperatorSlashedEvent(log *storage.TransactionLog) ([]*StateDiff, error) {
+	outputData, err := parseLogOutputForOperatorSlashedEvent(log.OutputData)
+	if err != nil {
+		return nil, err
+	}
+
+	stateDiffs := make([]*StateDiff, 0)
+
+	for i, strategy := range outputData.Strategies {
+		wadsSlashed, success := numbers.NewBig257().SetString(outputData.WadsSlashed[i].String(), 10)
+		if !success {
+			return nil, xerrors.Errorf("Failed to convert wadsSlashed to big.Int: %s", outputData.WadsSlashed[i])
+		}
+		stateDiffs = append(stateDiffs, &StateDiff{
+			Event: &SlashDiff{
```

Comment:
See my above comment. StateDiff taking a "generic" type event is an antipattern in Go

Reply:
https://github.com/Layr-Labs/sidecar/pull/68/commits/849bd0db77bd553611d6ccdf776de875882dc9c5